File: delivery/src/main/java/br/org/restaurantepopular/delivery/rest/api/user/IUserResource.java
package br.org.restaurantepopular.delivery.rest.api.user;

import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.concurrent.CompletionStage;

@RestController
@RequestMapping("/users")
    public interface IUserResource {

    @GetMapping(
            value = "{id}",
            produces = {MediaType.APPLICATION_JSON_VALUE},
            consumes = {MediaType.APPLICATION_JSON_VALUE})
    CompletionStage<UserDTO> getUserById(@PathVariable("id") String id);

    @PostMapping(
            produces = {MediaType.APPLICATION_JSON_VALUE},
            consumes = {MediaType.APPLICATION_JSON_VALUE})
    CompletionStage createUser(@RequestBody UserDTO userDTO);

    @PutMapping(
            produces = {MediaType.APPLICATION_JSON_VALUE},
            consumes = {MediaType.APPLICATION_JSON_VALUE})
    CompletionStage<ResponseEntity> updateUser(@RequestBody UserDTO userDTO);

    @GetMapping(produces = {MediaType.APPLICATION_JSON_VALUE},
                consumes = {MediaType.APPLICATION_JSON_VALUE})
    CompletionStage<List<UserDTO>> listUsers();


    @DeleteMapping(value = "{id}")
    CompletionStage<ResponseEntity> deleteUser(@PathVariable("id") String id);


    }

File: delivery/src/main/java/br/org/restaurantepopular/delivery/rest/api/user/IUserResource.kt
package br.org.restaurantepopular.delivery.rest.api.user

import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.util.concurrent.CompletionStage

@RestController
@RequestMapping("/users")
interface IUserResource {

    @GetMapping(
        value = ["{id}"],
        produces = [MediaType.APPLICATION_JSON_VALUE], consumes = [MediaType.APPLICATION_JSON_VALUE]
    )
    fun getUserById(@PathVariable("id") id: String): CompletionStage<UserDTO>

    @PostMapping(produces = [MediaType.APPLICATION_JSON_VALUE], consumes = [MediaType.APPLICATION_JSON_VALUE])
    fun createUser(@RequestBody userDTO: UserDTO): CompletionStage<*>

    @PutMapping(
        value = ["{id}"],
        produces = [MediaType.APPLICATION_JSON_VALUE], consumes = [MediaType.APPLICATION_JSON_VALUE]
    )
    fun updateUser(
        @PathVariable("id") id: String,
        @RequestBody userDTO: UserDTO
    ): CompletionStage<ResponseEntity<Unit>>

    @GetMapping(produces = [MediaType.APPLICATION_JSON_VALUE], consumes = [MediaType.APPLICATION_JSON_VALUE])
    fun listUsers(): CompletionStage<List<UserDTO>>

    @DeleteMapping(value = ["{id}"])
    fun deleteUser(@PathVariable("id") id: String): CompletionStage<ResponseEntity<Unit>>
}
